pkg/component/runtime/manager_test.go

```diff
@@ -1373,49 +1373,55 @@ func TestManager_FakeInput_GoodUnitToBad(t *testing.T) {
 					subErrCh <- fmt.Errorf("component failed: %s", state.Message)
 				} else {
 					unit, ok := state.Units[ComponentUnitKey{UnitType: client.UnitTypeInput, UnitID: "good-input"}]
-					if ok {
-						if unitGood {
-							if unit.State == client.UnitStateFailed {
-								subErrCh <- fmt.Errorf("unit failed: %s", unit.Message)
-							} else if unit.State == client.UnitStateHealthy {
-								// good unit it; now make it bad
-								t.Logf("marking good-input as having a hard-error for config")
-								updatedComp := comp
-								updatedComp.Units = make([]component.Unit, len(comp.Units))
-								copy(updatedComp.Units, comp.Units)
-								updatedComp.Units[1] = component.Unit{
-									ID:   "good-input",
-									Type: client.UnitTypeInput,
-									Err:  errors.New("hard-error for config"),
-								}
-								unitGood = false
-								err := m.Update(component.Model{Components: []component.Component{updatedComp}})
-								if err != nil {
-									subErrCh <- err
-								}
-							} else if unit.State == client.UnitStateStarting {
-								// acceptable
-							} else {
-								// unknown state that should not have occurred
-								subErrCh <- fmt.Errorf("unit reported unexpected state: %v", unit.State)
+					if !ok {
+						subErrCh <- errors.New("unit missing: good-input")
+						return
+					}
+					if unitGood {
+						if unit.State == client.UnitStateFailed {
```

Comment:
Looking at the comparisons being done in each of the `if` statements, maybe a `switch` would be even clearer?